src/codechicken/nei/recipe/GuiCraftingRecipe.java

```diff
@@ -6,22 +6,21 @@
 import net.minecraft.client.gui.inventory.GuiContainer;
 
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public class GuiCraftingRecipe extends GuiRecipe
 {
     public static boolean openRecipeGui(String outputId, Object... results) {
         Minecraft mc = NEIClientUtils.mc();
         GuiContainer prevscreen = mc.currentScreen instanceof GuiContainer ? (GuiContainer) mc.currentScreen : null;
 
-        TaskProfiler profiler = ProfilerRecipeHandler.getProfiler();
-        ArrayList<ICraftingHandler> handlers = new ArrayList<>();
-        for (ICraftingHandler craftinghandler : craftinghandlers) {
-            profiler.start(craftinghandler.getRecipeName());
-            ICraftingHandler handler = craftinghandler.getRecipeHandler(outputId, results);
-            if (handler.numRecipes() > 0)
-                handlers.add(handler);
-        }
-        profiler.end();
+        ArrayList<ICraftingHandler> handlers = craftinghandlers.parallelStream()
```

Comment:
Please add the profiler back, but just have it run surrounding the parallel stream, so we at least get some course metrics.

Reply:
Look at the other parallel stream I use, they keep a dedicated threadpool around -- I think we probably want to reuse the same threadpool if we can.  People had a non trivial amount of lag with the common pool.  https://github.com/GTNewHorizons/NotEnoughItems/commit/4b56f78b4685cb5c6135bfad9180ab569114262b is the commit that added it.